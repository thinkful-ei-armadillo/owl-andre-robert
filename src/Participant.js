import React, { Component } from 'react';
//import STORE from './store';
//import './ParticipantList.css';
//import Participant from './Participant';

class Participant extends Component {
  render() {
    return (
      <div className="participant">
        <p>{this.props.participant.name}</p>
        <img src = {this.props.participant.avatar} alt ="user avatar" ></img>
        <p>{(this.props.participant.inSession) ? 'in session' : 'not in session'}</p>
      </div>
    );
  }
}

export default Participant;
