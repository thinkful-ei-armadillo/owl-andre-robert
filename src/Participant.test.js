import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';
import { STORE } from './store';

describe('Testing Particpant generator', () => {
  // smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant  key={'gsdfgsdfg'} participant={STORE.participants[0]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Participant key={'gsdfgsdfg'} participant={STORE.participants[0]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});