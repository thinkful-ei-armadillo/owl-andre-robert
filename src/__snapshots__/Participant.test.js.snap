// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Particpant generator renders the UI as expected 1`] = `
<div
  className="participant"
>
  <p>
    Koren Templeton
  </p>
  <img
    alt="user avatar"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <p>
    in session
  </p>
</div>
`;
